### PR TITLE
Up to date Github actions and cache node_modules

### PR DESCRIPTION
**Pull request by McGiverGim**
This PR contains two commits:
- The first is an update of some actions (checkout, cache, setup-node, download-artifact, upload-artifact) to the latest version available.
- The second one, adds the node_modules to the cache with a Github action. I've done some tests, but is difficult to say how better it goes. The actions execution is very variable, the `yarn install` command can take from 25 seconds to 3 minutes, and using the cache it takes only 1 second, and downloading the cache from 5 seconds to 20 seconds. So it seems to go better always but the exact time is unknown.

We have 5 executions (one for the test, one for each OS build) so this makes better all of them.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Review by Asizon — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Pull request merged**
